check/src/kindcheck.rs

```diff
@@ -151,6 +151,7 @@ impl<'a> KindCheck<'a> {
 
     fn kindcheck(&mut self, typ: &TcType) -> Result<(RcKind, TcType)> {
         match **typ {
+            Type::Hole => Ok((self.subs.new_var(), Type::hole())),
```

Comment:
Any reason why you are not just calling typ.clone()? clarity?

Thought that may be better as with a cache it will still allocate a `Type::Hole` even if a hole is not used (and with clone() we are sharing a hole which definitely exists)